File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/Token.java
package com.cuupa.classificator.services.kb.semantic.token;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public abstract class Token {

	List<String> tokenValue;

	public abstract boolean match(String text);

	public List<String> getTokenValue() {
		return tokenValue;
	}

	void setTokenValue(List<String> tokenValue) {
		this.tokenValue = tokenValue;
	}

	public abstract int getDistance();

	@NotNull
	@Override
	protected abstract Token clone();

	@Override
	public String toString() {
		return String.join(",", tokenValue);
	}
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/token/Token.kt
package com.cuupa.classificator.services.kb.semantic.token

abstract class Token {
    @JvmField
    var tokenValue: MutableList<String> = mutableListOf()

    abstract fun match(text: String?): Boolean

    abstract val distance: Int
    abstract fun clone(): Token
    override fun toString(): String {
        return java.lang.String.join(",", tokenValue)
    }
}